pkg/workload/workload.go

```diff
@@ -156,12 +203,18 @@ func ResourceQuantity(name corev1.ResourceName, v int64) resource.Quantity {
 	}
 }
 
-func (r Requests) scale(f int64) {
+func (r Requests) scaleUp(f int64) {
 	for name := range r {
 		r[name] *= f
 	}
 }
 
+func (r Requests) scaleDown(f int64) {
+	for name := range r {
+		r[name] /= f
+	}
+}
```

Comment:
Are there any potential issues? What happens if `r[name] /= f` isn't divisible?

Reply:
The way it's used, the request sould always be a multiple f. 
In theory if it's not divisible it will just round down the result.

Reply:
Right. However, I think it is dangerous to assume that divisible values are in the receiver.

So, I would suggest this function returns a bool indicating whether all results are correct.

```suggestion
func (r Requests) scaleDown(f int64) bool {
	incorrect := false
	for name := range r {
		res := r[name]
		res /= f
		incorrect = (res*f != r[name])
		r[name] = res
	}
	return incorrect
}
```

Reply:
This means I don't want to hide the fact that the receiver must have divisible values.

Reply:
The values in admissions computed by multiplying the values for one pod with the number of pods ,  we only use this to divide to the same count. Sure we can  detect this and return it but what should we do abort it after that?

```suggestion
func (r Requests) scaleDown(f int64) bool {
        incorrect := false
	for name := range r {
		r[name] /= f
		incorrect |= (r[name] % f) > 0
	}
	return incorrect
}
```

Reply:
My concern is the use case other than L151, although those cases don't exist in this PR. So, in this case (L151), we don't need to check whether the result is incorrect.

This means for L151:

```go
// ignore check
_ = setRes.Requests.scaleDown(int64(setRes.Count))
```

My concern is for the future if this function is used other than L151.

Reply:
My suggestion might be too much. So let me know what others think. @alculquicondor

Reply:
Depending on how we go with https://github.com/kubernetes-sigs/kueue/pull/756#discussion_r1190638317, the scale down might only be needed for workloads that ware admitted with v0.3 and updated with v0.4.

Reply:
Yeah, I get the concern... but Workload is an object fully controlled by Kueue. And the Quantity object is not a floating point. It's a decimal implementation. So this is quite safe.

Maybe we can add in the Workload validation that the quantities are multiple of count?